src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -10,21 +10,47 @@ const CanonicalChartbeatBeacon = ({
   chartbeatUID,
   useCanonical,
   chartbeatSource,
-}) => (
-  <Helmet>
-    <script async type="text/javascript">
-      {`
+  hasCookie,
+  title,
+  referrer,
+  hasReferrer,
+}) => {
+  const optionalConfigValues = {
+    ...(hasReferrer && { virtualReferrer: referrer }),
+    ...(hasCookie && { idSync: { bbc_hid: cookie } }),
+  };
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.pSUPERFLY) {
+      window.pSUPERFLY.virtualPage({
+        uid: chartbeatUID,
+        domain,
+        useCanonical,
+        useCanonicalDomain: useCanonical,
+        title,
+        type,
+        sections,
+        ...optionalConfigValues,
+      });
+    }
+  });
+  return (
+    <Helmet>
+      <script async type="text/javascript">
+        {`
         (function(){
           var _sf_async_config = window._sf_async_config = (window._sf_async_config || {});
           _sf_async_config.uid = ${chartbeatUID};
           _sf_async_config.domain = "${domain}";
           _sf_async_config.useCanonical = ${useCanonical};
           _sf_async_config.useCanonicalDomain = ${useCanonical};
-          _sf_async_config.contentType = "${type}";
+          _sf_async_config.title = "${title}";
+          _sf_async_config.type = "${type}";
           _sf_async_config.sections = "${sections}";
-          _sf_async_config.idSync = {
-           bbc_hid: "${cookie}"
-          };
+          _sf_async_config = Object.assign(_sf_async_config, ${JSON.stringify(
+            optionalConfigValues,
+          )});
```

Comment:
This is a slight improvement, is there a reason we can't assemble the whole `_sf_async_config` this way? That would eliminate even more redundancy.